plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryTypeMapping.java

```diff
@@ -161,6 +166,34 @@ public void testNumericMapping()
+    {
+        // BigQuery views always return DECIMAL(38, 9)
+        SqlDataTypeTest.create()
+                .addRoundTrip("NUMERIC(3, 0)", "NUMERIC '193'", createDecimalType(38, 9), "CAST(193 AS DECIMAL(38, 9))")
+                .addRoundTrip("NUMERIC(3, 0)", "NUMERIC '19'", createDecimalType(38, 9), "CAST(19 AS DECIMAL(38, 9))")
+                .addRoundTrip("NUMERIC(3, 0)", "NUMERIC '-193'", createDecimalType(38, 9), "CAST(-193 AS DECIMAL(38, 9))")
+                .addRoundTrip("NUMERIC(3, 1)", "NUMERIC '10.0'", createDecimalType(38, 9), "CAST(10.0 AS DECIMAL(38, 9))")
+                .addRoundTrip("NUMERIC(3, 1)", "NUMERIC '10.1'", createDecimalType(38, 9), "CAST(10.1 AS DECIMAL(38, 9))")
+                .addRoundTrip("NUMERIC(3, 1)", "NUMERIC '-10.1'", createDecimalType(38, 9), "CAST(-10.1 AS DECIMAL(38, 9))")
+                .addRoundTrip("NUMERIC(4, 2)", "NUMERIC '2'", createDecimalType(38, 9), "CAST(2 AS DECIMAL(38, 9))")
+                .addRoundTrip("NUMERIC(4, 2)", "NUMERIC '2.3'", createDecimalType(38, 9), "CAST(2.3 AS DECIMAL(38, 9))")
+                .addRoundTrip("NUMERIC(24, 2)", "NUMERIC '2'", createDecimalType(38, 9), "CAST(2 AS DECIMAL(38, 9))")
+                .addRoundTrip("NUMERIC(24, 2)", "NUMERIC '2.3'", createDecimalType(38, 9), "CAST(2.3 AS DECIMAL(38, 9))")
+                .addRoundTrip("NUMERIC(24, 2)", "NUMERIC '123456789.3'", createDecimalType(38, 9), "CAST(123456789.3 AS DECIMAL(38, 9))")
+                .addRoundTrip("NUMERIC(24, 4)", "NUMERIC '12345678901234567890.31'", createDecimalType(38, 9), "CAST(12345678901234567890.31 AS DECIMAL(38, 9))")
+                .addRoundTrip("NUMERIC(29, 0)", "NUMERIC '27182818284590452353602874713'", createDecimalType(38, 9), "CAST('27182818284590452353602874713' AS DECIMAL(38, 9))")
+                .addRoundTrip("NUMERIC(29, 0)", "NUMERIC '-27182818284590452353602874713'", createDecimalType(38, 9), "CAST('-27182818284590452353602874713' AS DECIMAL(38, 9))")
+                .addRoundTrip("NUMERIC(30, 5)", "NUMERIC '3141592653589793238462643.38327'", createDecimalType(38, 9), "CAST(3141592653589793238462643.38327 AS DECIMAL(38, 9))")
+                .addRoundTrip("NUMERIC(30, 5)", "NUMERIC '-3141592653589793238462643.38327'", createDecimalType(38, 9), "CAST(-3141592653589793238462643.38327 AS DECIMAL(38, 9))")
+                .addRoundTrip("NUMERIC(38, 9)", "NUMERIC '100000000020000000001234567.123456789'", createDecimalType(38, 9), "CAST(100000000020000000001234567.123456789 AS DECIMAL(38, 9))")
+                .addRoundTrip("NUMERIC(38, 9)", "NUMERIC '-100000000020000000001234567.123456789'", createDecimalType(38, 9), "CAST(-100000000020000000001234567.123456789 AS DECIMAL(38, 9))")
+                .addRoundTrip("NUMERIC(10, 3)", "CAST(NULL AS NUMERIC)", createDecimalType(38, 9), "CAST(NULL AS DECIMAL(38, 9))")
+                .addRoundTrip("NUMERIC(38, 9)", "CAST(NULL AS NUMERIC)", createDecimalType(38, 9), "CAST(NULL AS DECIMAL(38, 9))")
```

Comment:
What happens to values which have > 9 decimal places? Can we have a test for that as documentation/verification of BigQuery behaviour?

Reply:
Table creation fails. Added a new test case.